zerver/lib/addressee.py

```diff
@@ -85,9 +85,8 @@ def user_profiles(self) -> List[UserProfile]:
         assert(self.is_private())
         return self._user_profiles  # type: ignore # assertion protects us
 
-    def stream_name(self) -> str:
+    def stream_name(self) -> Optional[str]:
         assert(self.is_stream())
-        assert(self._stream_name is not None)
```

Comment:
What's the story with this?

Reply:
Actually, I am not sure what the purpose of this assertion is. By the time someone calls `stream_name()`, the stream should already have been validated. Especially, now that an `Addressee` may have an ID and no name, this assertion is useless because we need a way to determine whether an ID or name was provided and that implies that `stream_name()` can return `None`.

We could also just choose to have a generic `stream()` method that returns a `Stream` object, and handles the work of choosing between the ID or name internally.

Reply:
I assume it was to make sure we ahve valid input.